### PR TITLE
Add comments to Sites.strings for translators

### PR DESCRIPTION
**Pull request by AliSoftware**
## Why?

While working on #18061 and making the `Sites.strings` strings used in `WordPressIntents` being part of the translation pipeline again, I forgot to add comments for translators for those strings like I did for the `InfoPlist.strings`.

This led to the strings not having any hint in GlotPress for translators to understand the context of what they were supposed to translate. (or worse, the first string picking up the `/* MARK: Sites.strings */` comment before it instead 😅 

<table><tr>
<td>image</td>
<td>image</td>
</tr></table>

## How?

- This PR simply adds `/* comments */` in front of each of those strings in the source `Sites.strings` file to help translators.
- I've also updated the `Localizable.strings` file (which merged the content of `Sites.strings` with key prefixes during code freeze, see #18061) with the exact same comments, so that those make it to GlotPress as soon as this lands in `trunk` (instead of waiting for the next code freeze and thus new merge of `Sites.strings` into `Localizable.strings` for them to be imported)

Since `git blame` tells me that @diegoreymendez is the original author of the `Sites.intentdefinition` files (in which those `Sites.strings` are used), I've also added him as reviewer so he can confirm the context provided by my comments are appropriate and give the right context to translators.

## Targeting 19.5

This PR is targeting the `release/19.5` branch and `19.5` milestone since those strings have been re-introduced in the `Localizable.strings` imported by GlotPress during 19.5 code freeze and we want those translations to ideally be part of 19.5

### CODE REVIEW

File: WordPress/WordPressIntents/en.lproj/Sites.strings
Comment (AliSoftware) [inline WordPress/WordPressIntents/en.lproj/Sites.strings]: @diegoreymendez Not sure when this one is used tbh, hence why I kept this vague.

I thought at first that it could be the text shown when you haven't selected any site yet, so I tried to add the Widget in my phone… but it got immediately pre-configured with the +platform9p2 site (not sure if that's because that's the default site I have set in WordPress, or because I previously configured the Widget when I tried the feature a while ago), so not sure that string would be for that case after all…?

File: WordPress/WordPressIntents/en.lproj/Sites.strings
Comment (AliSoftware) [inline WordPress/WordPressIntents/en.lproj/Sites.strings]: I tried playing with the iOS Widget and couldn't find a place anywhere in the widget's configuration screen where this string is used, so I think it might never be shown to the user after all? 🤔  cc @diegoreymendez

**Review by mokagio — APPROVED** [review]
LGTM!

Approving to unblock. It'd be great to wait for @diegoreymendez's input, but if that doesn't come that's good to merge. We can always open another one to remove stuff that might turn out to be unused.

---

> This PR is targeting the release/19.5 branch and 19.5 milestone since those strings have been re-introduced in the Localizable.strings imported by GlotPress during 19.5 code freeze and we want those translations to ideally be part of 19.5

🤔 . Would we also need to do a new `.po` generation after this lands in the release branch? Otherwise, the comments will go to GlotPress during the code freeze for 19.6.

No biggie either way. I just have my radar tuned up to merge as little as possible in the release branch and so pointed this out.

**Comment by AliSoftware** [comment]
> 🤔 . Would we also need to do a new .po generation after this lands in the release branch? Otherwise, the comments will go to GlotPress during the code freeze for 19.6.

Nope, the `AppStoreStrings.po` files are for metadata, while this is for app strings (`Localizable.strings`).
 - The [Release Notes and App Store Description](https://translate.wordpress.org/projects/apps/ios/release-notes/) GlotPress project imports its strings from `AppStoreStrings.po` file indeed, which in turn is constructed by assembling copies from `.txt` metadata files (not `.strings` files)
 - But the [Development](https://translate.wordpress.org/projects/apps/ios/dev/) GlotPress project imports its strings directly from the `Localizable.strings` file, without using an intermediate `.po` file

[Here, this is for you 😄 ☕](https://cdn3.vox-cdn.com/thumbor/5u-1qKEPqs6_N9HXdVPEKAjJN6k=/0x38:900x544/1600x900/cdn0.vox-cdn.com/uploads/chorus_image/image/43072140/starbucks-giant-coffee.0.0.jpg)

**Comment by mokagio** [comment]
> Nope, the AppStoreStrings.po files are for metadata, while this is for app strings (Localizable.strings).

🤦‍♂️ ☕

**Comment by AliSoftware** [comment]
FYI, discussed it quickly with @diegoreymendez in Slack, and he told me he won't have time to review those in details before Monday at best.

So we decided to merge this as is, to at least have some comments make their way to GlotPress before the polyglots start translating the strings without the context of widgets in mind, and before the originals (and their comments) are sent via the WeeklyKit to our translation vendor.

We can always revisit those strings and their translator hints later to refine them better.

File: WordPress/WordPressIntents/en.lproj/Sites.strings
Comment (AliSoftware) [reply WordPress/WordPressIntents/en.lproj/Sites.strings]: Confirmed with Diego in Slack: we agree that this is very likely not to be shown to end users at all.

In fact, even if it were, I doubt that the term "Intent" here would mean much to end users, considering that term comes from the technical name that Apple gives to those ways of configuring data to widgets (and Siri and such) via `.intentdefinition` files, a term that could speak to us devs but will not mean anything to end users.

For that reason, I've decided to remove that term from the `.strings` files (909570f) so that it gets removed from GlotPress (thus avoiding confusing polyglots not knowing how to translate it), and does not get translated for naught in the end.

**Pull request merged**
